src/modules/fancyzones/lib/JsonHelpers.cpp

```diff
@@ -137,798 +34,233 @@ namespace JSONHelpers
-        if (std::filesystem::exists(tmpFilePath))
-        {
-            try
-            {
-                if (auto customZoneSetJson = json::from_file(tmpFilePath); customZoneSetJson.has_value())
-                {
-                    if (auto customZoneSet = CustomZoneSetJSON::FromJson(customZoneSetJson.value()); customZoneSet.has_value())
-                    {
-                        customZoneSetsMap[customZoneSet->uuid] = std::move(customZoneSet->data);
-                    }
-                }
-            }
-            catch (const winrt::hresult_error&)
-            {
-                res = false;
-            }
+        json::JsonObject infoJson{};
+        infoJson.SetNamedValue(L"ref-width", json::value(canvasInfo.lastWorkAreaWidth));
+        infoJson.SetNamedValue(L"ref-height", json::value(canvasInfo.lastWorkAreaHeight));
 
-            DeleteTmpFile(tmpFilePath);
-        }
-        return res;
-    }
+        json::JsonArray zonesJson;
 
-    bool FancyZonesData::ParseDeletedCustomZoneSetsFromTmpFile(std::wstring_view tmpFilePath)
-    {
-        std::scoped_lock lock{ dataLock };
-        bool res = true;
-        if (std::filesystem::exists(tmpFilePath))
+        for (const auto& [x, y, width, height] : canvasInfo.zones)
         {
-            auto deletedZoneSetsJson = json::from_file(tmpFilePath);
-            try
-            {
-                auto deletedCustomZoneSets = deletedZoneSetsJson->GetNamedArray(L"deleted-custom-zone-sets");
-                for (auto zoneSet : deletedCustomZoneSets)
-                {
-                    std::wstring uuid = L"{" + std::wstring{ zoneSet.GetString() } + L"}";
-                    customZoneSetsMap.erase(std::wstring{ uuid });
-                }
-            }
-            catch (const winrt::hresult_error&)
-            {
-                res = false;
-            }
-
-            DeleteTmpFile(tmpFilePath);
+            json::JsonObject zoneJson;
+            zoneJson.SetNamedValue(L"X", json::value(x));
+            zoneJson.SetNamedValue(L"Y", json::value(y));
+            zoneJson.SetNamedValue(L"width", json::value(width));
+            zoneJson.SetNamedValue(L"height", json::value(height));
+            zonesJson.Append(zoneJson);
         }
-
-        return res;
+        infoJson.SetNamedValue(L"zones", zonesJson);
+        return infoJson;
     }
 
-    bool FancyZonesData::ParseAppZoneHistory(const json::JsonObject& fancyZonesDataJSON)
+    std::optional<FancyZonesDataTypes::CanvasLayoutInfo> CanvasLayoutInfoJSON::FromJson(const json::JsonObject& infoJson)
     {
-        std::scoped_lock lock{ dataLock };
         try
         {
-            auto appLastZones = fancyZonesDataJSON.GetNamedArray(L"app-zone-history");
+            FancyZonesDataTypes::CanvasLayoutInfo info;
+            info.lastWorkAreaWidth = static_cast<int>(infoJson.GetNamedNumber(L"ref-width"));
+            info.lastWorkAreaHeight = static_cast<int>(infoJson.GetNamedNumber(L"ref-height"));
```

Comment:
Replace all strings literals with const strings and explicitly mark them as non-localizable.